src/components/EditProfileModal/EditProfileModal.jsx

```diff
@@ -14,24 +18,72 @@ const EditProfileModal = ({ setModal, currentUser }) => {
         site: site,
     });
 
-    const editHandler = (e, userInput) => {
+    const [avatar, setAvatar] = useState('');
+
+    const editHandler = async (e, userInput) => {
         e.preventDefault();
+        if (avatar) {
+            const avatarFile = avatar;
+            const formData = new FormData();
+            formData.append('file', avatarFile);
+            formData.append('upload_preset', 'ti_social');
+            formData.append('folder', 'ti_social');
+
+            try {
+                await fetch(CLOUDINARY_URL, {
+                    method: 'POST',
+                    body: formData,
+                })
+                    .then((response) => response.json())
+                    .then((data) =>
+                        dispatch(
+                            editUserHelper({
+                                userInput: { ...userInput, avatarUrl: data.url },
+                                token,
+                            }),
+                        ),
+                    );
+            } catch (error) {
+                toast("Couldn't upload image.");
+                console.log(error);
+            }
+        } else {
+            dispatch(editUserHelper({ userInput, token }));
+        }
         setModal(false);
-        dispatch(editUserHelper({ userInput, token }));
     };
     return (
         <div className="py-4 border-[1px] bg-primary-bg w-[500px] z-10 border-color-grey rounded flex flex-col items-center justify-center">
             <i
-                class="fal fa-times w-full flex justify-end mr-8 hover:cursor-pointer"
+                className="fal fa-times w-full flex justify-end mr-8 hover:cursor-pointer"
                 onClick={() => setModal(false)}
             ></i>
             <p className="text-xl w-10/12 my-4">Edit your profile</p>
             <form
                 className="login-input-form w-10/12"
                 onSubmit={(event) => editHandler(event, userInput)}
             >
+                <div className="form-input my-4 flex flex-col items-center">
+                    <label className="avatar relative w-max cursor-pointer mx-auto my-2">
+                        <img
+                            src={'https://i.pravatar.cc/200'}
+                            alt="profile-avatar"
+                            className="rounded-full w-fit"
+                        />
```

Comment:
Suggestion: You can show the updated profile image on the edit profile modal by using `URL.createObjectURL`